core/scripts/vrfv2plus/testnet/util.go

```diff
@@ -0,0 +1,305 @@
+		stalenessSeconds,       // stalenessSeconds
+		gasAfterPayment,        // gasAfterPaymentCalculation
+		fallbackWeiPerUnitLink, // 0.01 eth per link fallbackLinkPrice
+		feeConfig,
+	)
+	helpers.PanicErr(err)
+	helpers.ConfirmTXMined(context.Background(), e.Ec, tx, e.ChainID)
+}
+
+func registerCoordinatorProvingKey(e helpers.Environment,
+	coordinator vrf_coordinator_v2plus.VRFCoordinatorV2Plus, uncompressed string, oracleAddress string) {
+	pubBytes, err := hex.DecodeString(uncompressed)
+	helpers.PanicErr(err)
+	pk, err := crypto.UnmarshalPubkey(pubBytes)
+	helpers.PanicErr(err)
+	tx, err := coordinator.RegisterProvingKey(e.Owner,
+		common.HexToAddress(oracleAddress),
+		[2]*big.Int{pk.X, pk.Y})
+	helpers.PanicErr(err)
+	helpers.ConfirmTXMined(
+		context.Background(),
+		e.Ec,
+		tx,
+		e.ChainID,
+		fmt.Sprintf("Uncompressed public key: %s,", uncompressed),
+		fmt.Sprintf("Oracle address: %s,", oracleAddress),
+	)
+}
+
+func getRlpHeaders(env helpers.Environment, blockNumbers []*big.Int) (headers [][]byte, err error) {
+	headers = [][]byte{}
+	for _, blockNum := range blockNumbers {
+		// Avalanche block headers are special, handle them by using the avalanche rpc client
+		// rather than the regular go-ethereum ethclient.
+		if helpers.IsAvaxNetwork(env.ChainID) {
+			// Get child block since it's the one that has the parent hash in its header.
+			h, err := env.AvaxEc.HeaderByNumber(
+				context.Background(),
+				new(big.Int).Set(blockNum).Add(blockNum, big.NewInt(1)),
+			)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get header: %+v", err)
+			}
+			// We can still use vanilla go-ethereum rlp.EncodeToBytes, see e.g
+			// https://github.com/ava-labs/coreth/blob/e3ca41bf5295a9a7ca1aeaf29d541fcbb94f79b1/core/types/hashing.go#L49-L57.
+			rlpHeader, err := rlp.EncodeToBytes(h)
+			if err != nil {
+				return nil, fmt.Errorf("failed to encode rlp: %+v", err)
+			}
+
+			// Sanity check - can be un-commented if storeVerifyHeader is failing due to unexpected
+			// blockhash.
+			//bh := crypto.Keccak256Hash(rlpHeader)
+			//fmt.Println("Calculated BH:", bh.String(),
+			//	"fetched BH:", h.Hash(),
+			//	"block number:", new(big.Int).Set(blockNum).Add(blockNum, big.NewInt(1)).String())
+
+			headers = append(headers, rlpHeader)
+		} else {
+			// Get child block since it's the one that has the parent hash in its header.
+			h, err := env.Ec.HeaderByNumber(
+				context.Background(),
+				new(big.Int).Set(blockNum).Add(blockNum, big.NewInt(1)),
+			)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get header: %+v", err)
+			}
+			rlpHeader, err := rlp.EncodeToBytes(h)
+			if err != nil {
+				return nil, fmt.Errorf("failed to encode rlp: %+v", err)
+			}
+
+			headers = append(headers, rlpHeader)
+		}
+	}
+	return
+}
+
+// binarySearch finds the highest value within the range bottom-top at which the test function is
+// true.
+func binarySearch(top, bottom *big.Int, test func(amount *big.Int) bool) *big.Int {
```

Comment:
Lets pull this out into the common/helpers module within core/scripts. We can also move the tests there.

Reply:
As a result of that you can also factor it out from the vrfv2/testnet scripts directory as well.

Reply:
Sure, doing that